src/main/kotlin/uk/gov/gdx/datashare/config/WebClientConfiguration.kt

```diff
@@ -22,19 +27,57 @@ class WebClientConfiguration(
 
   @Bean
   fun levApiWebClient(): WebClient {
-    val connectionProvider = ConnectionProvider.builder("levApi")
+    return WebClient.builder()
+      .baseUrl(levApiRootUri)
+      .clientConnector(ReactorClientHttpConnector(createHttpClient("levApi")))
+      .defaultHeader("X-Auth-Aud", "gdx-data-share")
+      .defaultHeader("X-Auth-Username", "gdx-data-share-user")
+      .build()
+  }
+
+  @Bean
+  @ConditionalOnProperty(name = ["api.base.prisoner-event.enabled"], havingValue = "true")
+  fun prisonerSearchApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId("prisoner-search")
+
+    return WebClient.builder()
+      .baseUrl(prisonerSearchApiUri)
+      .clientConnector(ReactorClientHttpConnector(createHttpClient("prisonerSearch")))
+      .filter(oauth2Client)
+      .codecs { codecs -> codecs.defaultCodecs().maxInMemorySize(2 * 1024 * 1024) }
+      .build()
+  }
+
+  private fun createHttpClient(connectionName: String): HttpClient {
+    val connectionProvider = ConnectionProvider.builder(connectionName)
       .maxConnections(100)
       .maxIdleTime(Duration.ofSeconds(20))
       .maxLifeTime(Duration.ofSeconds(60))
       .pendingAcquireTimeout(Duration.ofSeconds(60))
       .evictInBackground(Duration.ofSeconds(120))
       .build()
-    val httpClient = HttpClient.create(connectionProvider).responseTimeout(Duration.ofSeconds(10))
+
+    return HttpClient.create(connectionProvider).responseTimeout(Duration.ofMinutes(10))
```

Comment:
I think a 10 minute timeout here is pretty extreme? It's also incompatible with the other configured options I think

Reply:
Updated to 10s